File: javascript/src/models/eventOut.ts
// this file is @generated

export interface EventOut {
  /** The event type's name */
  eventType: string;
  payload: string;
  timestamp: Date;
}

export const EventOutSerializer = {
  _fromJsonObject(object: any): EventOut {
    return {
      eventType: object["eventType"],
      payload: object["payload"],
      timestamp: new Date(object["timestamp"]),
    };
  },

  _toJsonObject(self: EventOut): any {
    return {
      eventType: self.eventType,
      payload: self.payload,
      timestamp: self.timestamp,
    };
  },
};

File: rust/src/models/event_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct EventOut {
    /// The event type's name
    #[serde(rename = "eventType")]
    pub event_type: String,

    pub payload: String,

    pub timestamp: chrono::DateTime<chrono::Utc>,
}
